nncf/common/graph/graph.py

```diff
@@ -148,17 +148,23 @@ def __init__(
         self.parallel_input_port_ids = parallel_input_port_ids
 
     def __str__(self):
-        return str(self.from_node) + " -> " + str(self.tensor_shape) + " -> " + str(self.to_node)
+        return f"{self.from_node}:{self.output_port_id} -> {self.tensor_shape} -> {self.to_node}:{self.input_port_id}"
 
     def __hash__(self):
-        return hash(str(self))
+        return hash(
```

Comment:
So what's the difference here exactly?

Reply:
What is the definition of  equal edges? I believe two edges are equal iff all attributes are the same. Do you think it's wrong?

Reply:
It doesn't matter that much whether you are hashing the string representation of all the params or the params themselves. If you delegate the hashing to hashing the string, then you only have one place to adjust when adding/removing params, and that is the `__str__` method.

Reply:
Strange behavior encountered: in case hash is
```
hash(str(self.from_node) + " -> " + str(self.tensor_shape) + " -> " + str(self.to_node))
```
test `test_graph_analysis.test_graph_pattern_io_building` passed without a problem. But for hash
```
hash(f"{self.from_node}:{self.output_port_id} -> {self.tensor_shape} -> {self.to_node}:{self.input_port_id}")
```

the same test fails due to one of the edge has wrong input port_id (1 instead of 0)
I'm trying to investigate the root cause as there could be a potential bug in NNCFGraph code

Reply:
There is a bug in `test_graph_analysis.test_graph_pattern_io_building` references, I'll fix it

Reply:
Done